IRremoteESP8266.cpp

```diff
@@ -450,6 +450,47 @@ void ICACHE_FLASH_ATTR IRsend::sendDenon (unsigned long data,  int nbits) {
   ledOff();
 }
 
+// Gree, from https://github.com/ToniA/arduino-heatpumpir/blob/master/GreeHeatpumpIR.cpp
+void ICACHE_FLASH_ATTR IRsend::sendGree(unsigned char data[]) {
+
+  uint8_t i = 0;
+
+  // Set IR carrier frequency
+  enableIROut(38);
+
+  // Header #1
+  mark(GREE_HDR_MARK);
+  space(GREE_HDR_SPACE);
+
+  // Data #1
+  for (; i < 4; i++)
+    sendData(GREE_BIT_MARK, GREE_ONE_SPACE, GREE_BIT_MARK, GREE_ZERO_SPACE,
+             data[i], 8, false);
+
+  // Footer #2 (010)
+  mark(GREE_BIT_MARK);
+  space(GREE_ZERO_SPACE);
+  mark(GREE_BIT_MARK);
+  space(GREE_ONE_SPACE);
+  mark(GREE_BIT_MARK);
+  space(GREE_ZERO_SPACE);
+
+  // Header #2
+  mark(GREE_BIT_MARK);
+  space(GREE_MSG_SPACE);
+
+  // Data #2
+  for (; i < 8; i++)
+    sendData(GREE_BIT_MARK, GREE_ONE_SPACE, GREE_BIT_MARK, GREE_ZERO_SPACE,
+             data[i], 8, false);
+
+  // Footer #2
+  mark(GREE_BIT_MARK);
+  space(0);
```

Comment:
You don't need this. ledOff() will do.
However you probably should end with `space(GREE_MSG_SPACE);` instead of  `space(0);` or `ledOff;`